cli/azd/pkg/graphsdk/graph_client.go

```diff
@@ -0,0 +1,53 @@
+package graphsdk
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+type GraphClient struct {
+	pipeline runtime.Pipeline
+	host     string
+}
+
+// Creates a new instance of the Microsoft Graph client
+func NewGraphClient(
```

Comment:
Graph client has a very similar shape/style to the request builder style of the official SDK if we can ever move to it.